recipes/gdk-pixbuf/all/conanfile.py

```diff
@@ -0,0 +1,118 @@
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import shutil
+
+
+class LibnameConan(ConanFile):
+    name = "gdk-pixbuf"
+    description = "toolkit for image loading and pixel buffer manipulation"
+    topics = ("conan", "gdk-pixbuf", "image")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://developer.gnome.org/gdk-pixbuf/"
+    license = "LGPL-2.1-or-later"
+    generators = "pkg_config"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_libpng": [True, False],
+        "with_libtiff": [True, False],
+        "with_libjpeg": [True, False],
+        "with_jasper": [True, False],
+        }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_libpng": True,
+        "with_libtiff": True,
+        "with_libjpeg": True,
+        "with_jasper": False,
+        }
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.settings.os == "Macos":
+            # when running gdk-pixbuf-query-loaders
+            # dyld: malformed mach-o: load commands size (97560) > 32768
+            raise ConanInvalidConfiguration('This package does not support Macos currently')
+    
+    def build_requirements(self):
+        self.build_requires('meson/0.54.2')
+        if not tools.which('pkg-config') or self.settings.os == "Windows":
+            self.build_requires('pkgconf/1.7.3')
+    
+    def requirements(self):
+        self.requires('glib/2.67.0')
+        if self.options.with_libpng:
+            self.requires('libpng/1.6.37')
+        if self.options.with_libtiff:
+            self.requires('libtiff/4.0.9')
+        if self.options.with_libjpeg:
+            self.requires('libjpeg/9d')
+        if self.options.with_jasper:
+            self.requires('jasper/2.0.19')
```

Comment:
```suggestion
            self.requires('libjpeg/9d')
        if self.options.with_jasper:
            self.requires('jasper/2.0.19')
```

Is there any problem mixing with libjpeg-turbo? 

jasper.jpeg = ["libjpeg", "libjpeg-turbo"]

Reply:
I'll try this. I don't understand the code suggestion though

Reply:
usually we don't mix libjpeg and libjpeg-turbo (there is a related issue here in CCI). Thus you can use:

    conan install gdk-pixbuf/2.42.0@ -o jasper.jpeg=libjpeg-turbo

It will install libjpeg and libjpeg-turbo for the same package.

Reply:
Some cases:
https://github.com/conan-io/conan-center-index/issues/3330 
https://github.com/conan-io/conan-center-index/issues/1696
https://github.com/conan-io/conan-center-index/issues/2894

Reply:
it should be ok now with https://github.com/conan-io/conan-center-index/pull/3623/commits/e9b32fc07ac0614e9d125e5639978b7d1f2f6515